weighter.py

```diff
@@ -0,0 +1,159 @@
+    proc = subprocess.Popen(args)
+    proc.wait()
+
+
+def load_previous_results(tag):
+    '''
+    This function should load all the previous results from eos and calculate the mean over each muon.
+    '''
+    prev_results = []
+    prev_indeces = []
+    files = os.listdir(EOS_PATH)
+
+    i = 0
+    while True:
+        filename = "xs_" + tag + str(i) + ".npy"
+        if filename not in files:
+            break
+
+        xs = np.load(get_xs_path(tag, i))
+        prev_results.append(loss(np.array(xs)))
+
+        indeces = np.load(get_indeces_path(tag, i))
+        prev_indeces.append(indeces)
+        i += 1
+
+    return prev_results, prev_indeces
+
+
+
+
+def sample_muons(muon_loss, muon_indeces, share=0.05):
+    '''
+    Function sample the indexes of muons according to weights
+    '''
+    if share == None:
+        share = 0.05
+
+    cum_loss = np.sum(np.array(muon_loss), axis=0)
+    cum_indeces = np.zeros(len(cum_loss))
+    for index in muon_indeces:
+        cum_indeces[index] += 1
+
+    weights = cum_loss / cum_indeces
+    sample_size = int(len(weigths) * share)
+
+    if sample_size == 0:
+        raise
+
+    return np.random.choice(len(weights), size=sample_size, p=weigts/np.sum(weights), replace=True)
+
+def create_muons_files(filename_read, filename_write, indexes):
+    '''
+    Function takes all the muons, choose subsample according to indeces and saves it to filename_write
+    '''
+    f = r.TFile.Open(filename_read, 'read')
+    intuple = f.Get('pythia8-Geant4')
+    out = r.TFile.Open(filename_write, 'recreate')
+    outtuple = intuple.CloneTree(0)
+
+    i = 0
+    indexes_pointer = 0
+
+    for muon in intuple:
+        if i == indexes[indexes_pointer]:
+            outtuple.Fill(muon)
+            indexes_pointer += 1
+        if indexes_pointer == len(indexes):
+            break
+
+        i += 1
+
+    outtuple.Write()
+
+def count_muons(filename_read):
+    '''
+    This function calculates number of muons in the file.
+    '''
+    f = r.TFile.Open(filename_read, 'read')
+    intuple = f.Get('pythia8-Geant4')
+    i = 0
+
+    for muon in intuple:
+        i += 1
+
+    return i
+
+def get_command_line(SLAVE_CMD, args):
+    return SLAVE_CMD.format(**vars(args))
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-f',
+        '--input',
+        default='root://eoslhcb.cern.ch/'
+        '/eos/ship/data/Mbias/'
+        'pythia8_Geant4-withCharm_onlyMuons_4magTarget.root')
+    parser.add_argument('--results', default='results.json')
+    parser.add_argument('--hists', default='hists.root')
+    parser.add_argument('--params', required=True)
+    parser.add_argument('--point_id', required=True)
+    parser.add_argument('--tag', default="")
+    parser.add_argument('--seed', type=int, default=1)
+    parser.add_argument('--share_muons', type=float)
+
+    args = parser.parse_args()
+    args.xs_path = os.path.join(EOS_PATH, get_xs_path(args.tag, args.point_id))
```

Comment:
the slave can't write to EOS directly. EOS does not work like a normal file system (but can be mounted via fuse).